solutions/cses-2101.mdx

```diff
@@ -0,0 +1,163 @@
+---
+id: cses-2101
+source: CSES
+title: New Roads Queries
+author: Dustin Miao
+---
+
+Let's construct a graph where edges have weights equal to the time they are added. The problem boils down to efficiently finding the path with the minimum maximum edge for each query. 
+
+Let's consider a single query between nodes $u, v$. One way of approaching this is to sort the edge weights from smallest to largest, and then add edges one by one until $u$ and $v$ are connected. If we use a [disjoint set union](/gold/dsu), the time complexity for each query is $\mathcal{O}(m \cdot \alpha(n))$, which is too slow.
+
+However, this hints at a faster solution. Note that our slow approach bears many similarities with [Kruskal's MST algorithm](/gold/mst/#kruskals), which implies that the path between $u$ and $v$ on any minimum spanning tree will minimize the maximum edge weight. For a more formal proof, see [here](https://www.geeksforgeeks.org/minimum-bottleneck-spanning-treembst/). 
+****
+All that is left to do is to efficiently query for the minimum node on a tree path. One way to do this is with [binary jumping](/plat/binary-jump). 
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(m \cdot \alpha(n) + n\log n)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+	
+const int MAXN = 2e5;
+const int LOGN = 18;
```

Comment:
still add a comment for this

Reply:
why?????

Reply:
log what? base what?
just say like `// log of MAXN base 2`